MeshLib/Elements/HexRule8.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "HexRule8.h"
+
+#include "logog/include/logog.hpp"
+
+#include "GeoLib/AnalyticalGeometry.h"
+
+#include "MeshLib/Node.h"
+#include "Quad.h"
+#include "Line.h"
+
+namespace MeshLib {
+
+const unsigned HexRule8::n_all_nodes;
+
+const unsigned HexRule8::n_base_nodes;
+
+const unsigned HexRule8::face_nodes[6][4] =
+{
+	{0, 3, 2, 1}, // Face 0
+	{0, 1, 5, 4}, // Face 1
+	{1, 2, 6, 5}, // Face 2
+	{2, 3, 7, 6}, // Face 3
+	{3, 0, 4, 7}, // Face 4
+	{4, 5, 6, 7}  // Face 5
+};
+
+const unsigned HexRule8::edge_nodes[12][2] =
+{
+	{0, 1}, // Edge 0
+	{1, 2}, // Edge 1
+	{2, 3}, // Edge 2
+	{0, 3}, // Edge 3
+	{4, 5}, // Edge 4
+	{5, 6}, // Edge 5
+	{6, 7}, // Edge 6
+	{4, 7}, // Edge 7
+	{0, 4}, // Edge 8
+	{1, 5}, // Edge 9
+	{2, 6}, // Edge 10
+	{3, 7}  // Edge 11
+};
+
+const Element* HexRule8::getFace(const Element* e, unsigned i)
+{
+	if (i < n_faces)
+	{
+		unsigned nFaceNodes (getNFaceNodes(i));
+		Node** nodes = new Node*[nFaceNodes];
+		for (unsigned j=0; j<nFaceNodes; j++)
+			nodes[j] = const_cast<Node*>(e->getNode(face_nodes[i][j]));
+		return new Quad(nodes);
+	}
+	ERR("Error in MeshLib::Element::getFace() - Index %d does not exist.", i);
+	return nullptr;
+}
+
+double HexRule8::computeVolume(Node const* const* _nodes)
+{
+	return GeoLib::calcTetrahedronVolume(_nodes[4]->getCoords(), _nodes[7]->getCoords(), _nodes[5]->getCoords(), _nodes[0]->getCoords())
```

Comment:
the PR is okay, but I am confident that the compiler will do something clever here. getcoords is a const method that just returns a value, so it is very likely this will be replaced here during compiling.